MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m

```diff
@@ -165,38 +165,36 @@ - (void)renewCurrentFileForStorageKey:(NSString *)storageKey {
   MSStorageBucket *bucket = [self bucketForStorageKey:storageKey];
   NSDate *creationDate = [NSDate date];
   NSString *fileId = MS_UUID_STRING;
-  NSString *filePath = [self filePathForStorageKey:storageKey logsId:fileId];
-  MSFile *file = [[MSFile alloc] initWithPath:filePath fileId:fileId creationDate:creationDate];
+  NSURL *fileURL = [self fileURLForStorageKey:storageKey logsId:fileId];
+  MSFile *file = [[MSFile alloc] initWithURL:fileURL fileId:fileId creationDate:creationDate];
   bucket.currentFile = file;
   [bucket.currentLogs removeAllObjects];
 }
 
-- (NSString *)directoryPathForStorageKey:(NSString *)storageKey {
-  NSString *filePath = [self.baseDirectoryPath stringByAppendingPathComponent:storageKey];
+- (NSURL *)directoryURLForStorageKey:(NSString *)storageKey {
+  NSURL *fileURL = [self.baseDirectoryURL URLByAppendingPathComponent:storageKey];
 
-  return filePath;
+  return fileURL;
 }
 
-- (NSString *)filePathForStorageKey:(NSString *)storageKey logsId:(nonnull NSString *)logsId {
+- (NSURL *)fileURLForStorageKey:(NSString *)storageKey logsId:(nonnull NSString *)logsId {
   NSString *fileName = [logsId stringByAppendingPathExtension:kMSFileExtension];
-  NSString *filePath = [[self directoryPathForStorageKey:storageKey] stringByAppendingPathComponent:fileName];
+  NSURL *fileURL = [[self directoryURLForStorageKey:storageKey] URLByAppendingPathComponent:fileName];
 
-  return filePath;
+  return fileURL;
 }
 
-- (NSString *)baseDirectoryPath {
-  if (!_baseDirectoryPath) {
-    NSString *appSupportPath =
-            [[NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES) lastObject]
-                    stringByStandardizingPath];
-    if (appSupportPath) {
-      _baseDirectoryPath = [appSupportPath stringByAppendingPathComponent:kMSLogsDirectory];
+- (NSURL *)baseDirectoryURL {
+  if (!_baseDirectoryURL) {
+    NSURL *appSupportURL = [[[NSFileManager defaultManager] URLsForDirectory:NSApplicationSupportDirectory inDomains:NSUserDomainMask] lastObject];
+    if (appSupportURL) {
+      _baseDirectoryURL = (NSURL * _Nonnull)[appSupportURL URLByAppendingPathComponent:kMSLogsDirectory];
     }
 
-    MSLogVerbose([MSMobileCenter logTag], @"Storage Path:\n%@", _baseDirectoryPath);
+    MSLogVerbose([MSMobileCenter logTag], @"Storage Path:\n%@", self->_baseDirectoryURL);
```

Comment:
Yep but it's not the only var accessed directly, if we go that way we should propertify them all in this class.